internal/argotunnel/translator.go

```diff
@@ -185,10 +185,14 @@ func (t *syncTranslator) getRouteFromIngress(ing *v1beta1.Ingress) (r *tunnelRou
 		secret := func() *resource {
 			if r, ok := hostsecret[rule.Host]; ok {
 				return r
-			} else if t.options.secret != nil {
-				return t.options.secret
 			}
-			return nil
+
+			secret, err := t.findVerifiedCertSecret(t.options.secrets, host)
```

Comment:
here the label related issues will become more apparent. Find is directly tied to the order of the secrets provided as options the command-line.  More that one secret might satisfy the host verification match, though only the first will be used.  If the user configures wildcard certs followed by specific certs, the specific cert will never be used. This is similar to the handling with labels, though better because the order is at least stable and controllable.